src/player.js

```diff
@@ -1157,10 +1157,34 @@ export default class Player extends FakeEventTarget {
       this._eventManager.listen(this, Html5Events.RATE_CHANGE, () => {
         this._playbackAttributesState.rate = this.playbackRate;
       });
+      this._eventManager.listen(this, CustomEvents.ENTER_FULLSCREEN, () => {
+        this._resetTextCuesAndReposition();
+      });
+      this._eventManager.listen(this, CustomEvents.EXIT_FULLSCREEN, () => {
+        this._resetTextCuesAndReposition();
+      });
     }
   }
 
   /**
+   * Reset the active cues hasBeenReset = true and then reposition it, timeout here is for the screen to
+   * finish render the fullscreen
+   * @returns {void}
+   * @private
+   */
+  _resetTextCuesAndReposition(): void {
+    this._updateTextDisplay([]);
+    for (let i = 0; i < this._activeTextCues.length; i++) {
+      this._activeTextCues[i].hasBeenReset = true;
+    }
+
+    // waiting the browser to render the screen
+    setTimeout(() => {
```

Comment:
save timeout id and handle cases when events come in consecutive manner, e.g. debounce it.
Also, move timeout to const value.

Reply:
@OrenMe i didn't debounce it - just ignored previous timeouts (the last one will be the one that will be executed - i think that's the behavior we want)